tiatoolbox/annotation/storage.py

```diff
@@ -1071,80 +1072,147 @@ def to_geodict(self) -> Dict[str, Any]:
+            relative_to [float, float]:
+                The x and y coordinates to use as the origin for the annotations.
+
+        Returns:
+            AnnotationStore:
+                A new annotation store with the annotations loaded from the file.
+
+        """
+        store = cls()
+        store.add_from_geojson(full_path, scale_factor, relative_to=relative_to)
+        return store
+
+    def add_from_geojson(
+        self,
+        full_path: Union[IO, str],
+        scale_factor: float = 1,
+        relative_to: Tuple[float, float] = None,
+    ) -> None:
+        """Add annotations from a .geojson file to an existing store. Make
+        a best effort to create valid shapely geometries from provided contours.
+
+        Args:
+            full_path (Union[IO, str, Path]):
+                The file path or handle to load from.
+            scale_factor (float):
+                The scale factor to use when loading the annotations. All coordinates
+                will be multiplied by this factor to allow import of annotations saved
+                at non-baseline resolution.
+            relative_to [float, float]:
+                The x and y coordinates to use as the origin for the annotations.
+
+        """
+
+        def transform_geom(geom):
+            """Helper function to transform a geometry if needed."""
+            if relative_to is not None:
+                # transform coords to be relative to given pt.
+                geom = translate(geom, -relative_to[0], -relative_to[1])
+            if scale_factor != 1:
+                geom = scale(
+                    geom,
+                    xfact=scale_factor,
+                    yfact=scale_factor,
+                    origin=(0, 0, 0),
+                )
+            return geom
+
+        geojson = self._load_cases(
+            full_path=full_path,
             string_fn=json.loads,
             file_fn=json.load,
         )
-        store = cls()
-        for feature in geojson["features"]:
-            geometry = feature2geometry(feature["geometry"])
-            properties = feature["properties"]
-            store.append(Annotation(geometry, properties))
-        return store
 
-    def to_geojson(self, fp: Optional[IO] = None) -> Optional[str]:
+        annotations = [
+            Annotation(
+                transform_geom(
+                    feature2geometry(feature["geometry"]),
+                ),
+                feature["properties"],
+            )
+            for feature in geojson["features"]
+        ]
+
+        print(f"added {len(annotations)} annotations")
+        self.append_many(annotations)
+
+    def to_geojson(self, full_path: Optional[IO] = None) -> Optional[str]:
         """Serialise the store to geoJSON.
 
         For more information on the geoJSON format see:
         - https://geojson.org/
         - https://tools.ietf.org/html/rfc7946
 
         Args:
-             fp (IO):
+             full_path (IO):
```

Comment:
This name doesn't make sense. It is not for a path but a file-like IO object. 
`fp` was chosen to be consistent with the Python standard library e.g. json: https://github.com/python/cpython/blob/c7662420d62e04eefd99d7cdbcc74f72e68f3a6a/Lib/json/__init__.py#L120

Although admittedly this also works with a file path. and should be a union of `IO` and `pathlib.Path`.

Full path also doesn't make sense as it works with non-full paths i.e. relative paths.